### PR TITLE
Add: New GitHubRESTApi class for querying GitHub API

### PR DESCRIPTION
**Pull request by bjoernricks**
**What**:

Add an API for querying the GitHub REST API

**Why**:

Provide an easy to use API for GitHub

**How**:

**Checklist**:

- [x] Tests
- [x] Conventional commit message
- [x] Documentation

### CODE REVIEW

File: pontos/github/api.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+        Get data of a GitHub release by tag
+
+        Args:
+            repo: GitHub repository (owner/name) to use
+            tag: The git tag for the release
+
+        Raises:
+            HTTPError if the request was invalid
+        """
+        api = f"/repos/{repo}/releases/tags/{tag}"
```
Comment (y0urself) [inline pontos/github/api.py:284]: Use `release_exists` in `release` to check if the release exists beforehand?

File: pontos/github/api.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+    def request(
+        self,
+        api: str,
+        *,
+        params: Optional[Dict[str, str]] = None,
+        data: Optional[Dict[str, str]] = None,
+        request: Optional[Callable] = None,
+    ) -> requests.Response:
+        headers = {
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply pontos/github/api.py:115]: I wasn't sure about that. Maybe you are right. I'll change that.

File: pontos/github/api.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            it = iter(self)
+            while True:
+                next(it)
+        except StopIteration:
+            pass
+
+
+def download(
+    url: str,
+    destination: Path,
```
[the remark this answers is not in the document]
Comment (bjoernricks) [reply pontos/github/api.py:66]: it's a bit more difficult. we still need to be able to return the action path where the file is stored at the end. I'll try to come up with a solution.

File: pontos/github/api.py (hunk 1 above)
Comment (bjoernricks) [reply pontos/github/api.py:284]: Not sure. Isn't this a tasks for the API user?

Comment (y0urself) [reply pontos/github/api.py:284]: I think this would be discussable ...

Comment (bjoernricks) [reply pontos/github/api.py:297]: An API user might know if a release already exists or not. Personally I don't see much value in checking it beforehand.

Comment (y0urself) [reply pontos/github/api.py:297]: Ok for me!

**Review by y0urself — APPROVED** [review]

**Review by y0urself — APPROVED** [review]

**Pull request merged**